CMakeLists.txt

```diff
@@ -753,54 +753,66 @@ if(HPX_WITH_AGAS_DUMP_REFCNT_ENTRIES)
   hpx_add_config_define(HPX_HAVE_AGAS_DUMP_REFCNT_ENTRIES)
 endif()
 
-# Should networking be supported?
-hpx_option(HPX_WITH_NETWORKING BOOL
-  "Enable support for networking and multi-node runs (default: ON)"
-  ON CATEGORY "Parcelport")
-if(HPX_WITH_NETWORKING)
-  hpx_add_config_define(HPX_HAVE_NETWORKING)
-
-  ## Parcelport related build options
-  set(_parcel_profiling_default OFF)
-  if(HPX_WITH_APEX)
-    set(_parcel_profiling_default ON)
-  endif()
-
-  hpx_option(HPX_WITH_PARCEL_PROFILING BOOL
-    "Enable profiling data for parcels"
-    ${_parcel_profiling_default} CATEGORY "Parcelport" ADVANCED)
-
-  if(HPX_WITH_PARCEL_PROFILING)
-    hpx_add_config_define(HPX_HAVE_PARCEL_PROFILING)
-  endif()
-
-  ## Parcelport related build options
-  hpx_option(HPX_WITH_PARCELPORT_LIBFABRIC BOOL
-    "Enable the libfabric based parcelport. This is currently an experimental feature"
-    OFF CATEGORY "Parcelport" ADVANCED)
-  hpx_option(HPX_WITH_PARCELPORT_VERBS BOOL
-    "Enable the ibverbs based parcelport. This is currently an experimental feature"
-    OFF CATEGORY "Parcelport" ADVANCED)
-  hpx_option(HPX_WITH_PARCELPORT_MPI BOOL
-    "Enable the MPI based parcelport."
-    OFF CATEGORY "Parcelport")
-  hpx_option(HPX_WITH_PARCELPORT_TCP BOOL
-    "Enable the TCP based parcelport."
+hpx_option(HPX_WITH_EXPERIMENTAL_LOCAL_RUNTIME BOOL
+  "Enable the experimental local runtime (this disables all distributed features of HPX, including components and actions) (default: OFF)"
+  OFF CATEGORY "Build Targets" ADVANCED)
+
+if(HPX_WITH_LOGGING AND HPX_WITH_EXPERIMENTAL_LOCAL_RUNTIME)
+  hpx_error("HPX_WITH_EXPERIMENTAL_LOCAL_RUNTIME=ON requires HPX_WITH_LOGGING=OFF")
+endif()
+
+if(NOT HPX_WITH_EXPERIMENTAL_LOCAL_RUNTIME)
+  hpx_add_config_define(HPX_HAVE_DISTRIBUTED_RUNTIME)
```

Comment:
May I suggest to use consistent names here? Either use `HPX_WITH_DISTRIBUTED_RUNTIME` or `HPX_HAVE_EXPERIMENTAL_LOCAL_RUNTIME`. I like the first one better, FWIW.

Reply:
Yeah, probably a good idea. I wanted to have `EXPERIMENTAL` in the CMake option, but I didn't want to have `HPX_WITH_EXPERIMENTAL_DISTRIBUTED_RUNTIME` because that's a lie. Perhaps I'll just make it clear in the description of the option that turning it off is experimental.

Reply:
I updated it to be `HPX_WITH_DISTRIBUTED_RUNTIME`.